server/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java

```diff
@@ -46,6 +46,9 @@ public static Logger getLogger(String prefix, Class<?> clazz) {
     }
 
     public static Logger getLogger(String prefix, Logger logger) {
+        if (prefix == null || prefix.length() == 0) {
```

Comment:
Why not fail here? These would just be noop calls that could be removed right?

Reply:
Right now everything gets routed through this method even if there isn't a prefix. How about this: I promise to remove as many of the `getLogger` variants as I can in a followup but I keep this like this in this PR so I don't make the change larger than it already is?

Reply:
Sure. Can you at least add a comment here that this leniency should/will be removed in the future?

Reply:
Sure!